tests/benchmarks/multicore/idle_ipc/src/main.c

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/pm/device.h>
+#include <zephyr/pm/device_runtime.h>
+#include <string.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/ipc/ipc_service.h>
+
+#ifdef CONFIG_TEST_EXTRA_STACK_SIZE
+#define STACKSIZE (1024 + CONFIG_TEST_EXTRA_STACK_SIZE)
+#else
+#define STACKSIZE (1024)
+#endif
+#define NUMBER_OF_MESSAGES_TO_SEND 20
+#define IPC_SEND_DEAD_TIME_MS	   1
+
+K_THREAD_STACK_DEFINE(ipc0_stack, STACKSIZE);
+
+LOG_MODULE_REGISTER(host, LOG_LEVEL_INF);
+
+static const struct device *const console_dev = DEVICE_DT_GET(DT_CHOSEN(zephyr_console));
+
+struct payload {
+	unsigned long cnt;
+	unsigned long size;
+	uint8_t data[];
+};
+
+struct payload *p_payload;
+
+static K_SEM_DEFINE(bound_sem, 0, 1);
+
+static void ep_bound(void *priv)
+{
+	k_sem_give(&bound_sem);
+}
+
+static void ep_recv(const void *data, size_t len, void *priv)
+{
+	uint8_t received_val = *((uint8_t *)data);
+	static uint8_t expected_val;
+
+	if ((received_val != expected_val) || (len != CONFIG_APP_IPC_SERVICE_MESSAGE_LEN)) {
+		printk("Unexpected message received_val: %d , expected_val: %d\n", received_val,
+		       expected_val);
+		printk("Test should not enter here\n");
+		__ASSERT_NO_MSG(1 == 0);
+	}
+
+	expected_val++;
+}
+
+static struct ipc_ept_cfg ep_cfg = {
+	.name = "ep0",
+	.cb = {
+			.bound = ep_bound,
+			.received = ep_recv,
+		},
+};
+
+int main(void)
+{
+	const struct device *ipc0_instance;
+	struct ipc_ept ep;
+	int ret;
+	unsigned long last_cnt = 0;
+	unsigned long delta = 0;
+
+	p_payload = (struct payload *)k_malloc(CONFIG_APP_IPC_SERVICE_MESSAGE_LEN);
+	if (!p_payload) {
+		printk("k_malloc() failure\n");
+		return -ENOMEM;
```

Comment:
please do assert here, as exit from main will lead to s2ram, thus it is hard to find what was the problem